sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/JacksonAdapter.java

```diff
@@ -46,19 +49,44 @@
      * Creates a new JacksonAdapter instance with default mapper settings.
      */
     public JacksonAdapter() {
-        this.simpleMapper = ObjectMapperShim.createSimpleMapper();
+        this((outerMapper, innerMapper) -> { });
+    }
+
+    /**
+     * Creates a new JacksonAdapter instance with default mapper settings and additional
+     * configuration.
+     *
+     * @param configure Function that applies additional configuration to outer
+     *                  mapper using inner (simple) mapper for module chaining.
+     */
+    public JacksonAdapter(BiConsumer<ObjectMapper, ObjectMapper> configure) {
```

Comment:
Since this is going to be the new public constructor, we should add a bit more details in the javadoc to describe what inner and outer mappers are here. Also, we should rename `configure` to something more descriptive.